src/react/components/common/tagInput/tagInput.test.tsx

```diff
@@ -0,0 +1,267 @@
+
+describe("Tag Input Component", () => {
+
+    function createComponent(props?: ITagInputProps):
+        ReactWrapper<ITagInputProps, ITagInputState> {
+        return mount(<TagInput {...(props || createProps())} />);
+    }
+
+    function createProps(tags?: ITag[], onChange?): ITagInputProps {
+        return {
+            tags: tags || MockFactory.createTestTags(),
+            lockedTags: [],
+            selectedRegions: [MockFactory.createTestRegion()],
+            onChange: onChange || jest.fn(),
+            onLockedTagsChange: jest.fn(),
+            onTagClick: jest.fn(),
+            onCtrlTagClick: jest.fn(),
+        };
+    }
+
+    it("Renders correctly", () => {
+        const tags = MockFactory.createTestTags();
+        const wrapper = createComponent(createProps(tags));
+        expect(wrapper.exists(".tag-input-toolbar")).toBe(true);
+        expect(wrapper.find(".tag-item-block").length).toBe(tags.length);
+    });
+
+    it("Calls onClick handler when clicking color box", () => {
+        const props: ITagInputProps = createProps();
+        const wrapper = createComponent(props);
+        wrapper.find(".tag-color").first().simulate("click");
+        expect(props.onTagClick).toBeCalledWith(props.tags[0]);
+        expect(wrapper.state().clickedColor).toBe(true);
+        expect(props.onCtrlTagClick).not.toBeCalled();
+    });
+
+    it("Calls onClick handler when clicking text", () => {
+        const props: ITagInputProps = createProps();
+        const wrapper = createComponent(props);
+        wrapper.find(".tag-name-text").first().simulate("click");
+        expect(props.onTagClick).toBeCalledWith(props.tags[0]);
+        expect(props.onCtrlTagClick).not.toBeCalled();
+    });
+
+    it("Calls onCtrlClick handler when clicking color box", () => {
+        const props: ITagInputProps = createProps();
+        const wrapper = createComponent(props);
+        wrapper.find(".tag-color").first().simulate("click", { ctrlKey: true });
+        expect(props.onCtrlTagClick).toBeCalledWith(props.tags[0]);
+        expect(wrapper.state().clickedColor).toBe(true);
+        expect(props.onTagClick).not.toBeCalled();
+    });
+
+    it("Calls onClick handler when clicking text", () => {
+        const props: ITagInputProps = createProps();
+        const wrapper = createComponent(props);
+        wrapper.find(".tag-name-text").first().simulate("click", { ctrlKey: true });
+        expect(props.onCtrlTagClick).toBeCalledWith(props.tags[0]);
+        expect(props.onTagClick).not.toBeCalled();
+    });
+
+    it("Adds a tag", () => {
+        const props: ITagInputProps = {
+            ...createProps(),
+            showTagInputBox: true,
+        };
+        const wrapper = createComponent(props);
+        const newTagName = "New Tag";
+        wrapper.find(".tag-input-box").simulate("keydown", { key: "Enter", target: { value: newTagName } });
+        expect(props.onChange).toBeCalled();
```

Comment:
Should you be checking some tag list here to see if the tag has been added instead of just checking if the onChange function has been called?

Reply:
Yep! Good call. I'll add that